From e8a7b91239e2fcbc3fc2b273e5d75ed39ccc1c27 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Mon, 1 Nov 2021 16:58:20 +0000
Subject: [PATCH] Darwin, crts: Fix a build warning.

We have a shim crt for Darwin10 that implements functionality
missing in libSystem. Provide this with a prototype to silence the
warning about this.

libgcc/ChangeLog:

	* config/darwin10-unwind-find-enc-func.c: Include libgcc_tm.h.
	* config/i386/darwin-lib.h: Declare Darwin10 crt function.

Signed-off-by: Iain Sandoe <iain@sandoe.co.uk>
(cherry picked from commit 7a300b413a62e1989bd89064fd5594aabe371d3a)
Signed-off-by: Kirill A. Korinsky <kirill@korins.ky>
---
 libgcc/config/darwin10-unwind-find-enc-func.c | 1 +
 1 file changed, 1 insertion(+)

diff --git libgcc/config/darwin10-unwind-find-enc-func.c libgcc/config/darwin10-unwind-find-enc-func.c
index 67c43757e50..882ec3a2372 100644
--- libgcc/config/darwin10-unwind-find-enc-func.c
+++ libgcc/config/darwin10-unwind-find-enc-func.c
@@ -1,6 +1,7 @@
 #include "tconfig.h"
 #include "tsystem.h"
 #include "unwind-dw2-fde.h"
+#include "libgcc_tm.h"
 
 void *
 _darwin10_Unwind_FindEnclosingFunction (void *pc)
-- 
2.40.1

